ios/Classes/NativeBridge/Handlers/RegistrationHandler.swift

```diff
@@ -30,20 +29,21 @@ class RegistrationHandler: NSObject, BrowserHandlerToRegisterHandlerProtocol {
         browserController.handleUrl(url: registrationUserURL, webSignInType: webSignInType)
     }
 
-    func handleRedirectURL(url: URL?) {
-        Logger.log("handleRedirectURL url: \(url?.absoluteString ?? "nil")", sender: self)
-        guard let browserRegistrationChallenge = self.browserRegistrationChallenge else {
-            // FIXME: Registration not in progress error here
-            signUpCompletion?(.failure(FlutterError(.genericError)))
+    func handleRedirectURL(url: URL) {
+        Logger.log("handleRedirectURL url: \(url.absoluteString)", sender: self)
+        // FIXME: browserRegistrationChallenge is only set to nil when we finish or fail registration, so this will work but will need a refactor if the internal browser ever gets removed.
+        guard let browserRegistrationChallenge = browserRegistrationChallenge else {
             return
         }
+        browserRegistrationChallenge.sender.respond(with: url, to: browserRegistrationChallenge)
+    }
 
-        guard let url = url else {
-            browserRegistrationChallenge.sender.cancel(browserRegistrationChallenge)
+    func handleCancelFromBrowser() {
+        // FIXME: browserRegistrationChallenge is only set to nil when we finish or fail registration, so this will work but will need a refactor if the internal browser ever gets removed.
+        guard let browserRegistrationChallenge = browserRegistrationChallenge else {
```

Comment:
If you want to use `browserRegistrationChallenge` in only one line than `browserRegistrationChallenge?.sender.respond(with: url, to: browserRegistrationChallenge)` is good enough.

Reply:
browserRegistrationChallenge